# Разбивка на одноклассовые классификации (LB - 0.818)
Метрика задачи позволяет переформулировать задачу из multi-label классификации и получить более мелкие задачи классификации, вплоть до 41 задачи одноклассовой классификации. Задачи одноклассовой классификации можно решать независимо, разделив их между участниками команды и/или между компьютерами.

In [22]:
conda create -n pycaret_env python=3.10 -y

Retrieving notices: done
Channels:
 - defaults
Platform: osx-64
Solving environment: done


==> WARNING: A newer version of conda exists. <==
    current version: 25.5.1
    latest version: 25.7.0

Please update conda by running

    $ conda update -n base -c defaults conda



## Package Plan ##

  environment location: /opt/anaconda3/envs/pycaret_env

  added / updated specs:
    - python=3.10


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    openssl-3.0.17             |       hee2dfae_0         4.6 MB
    pip-26.0.1                 |     pyhc872135_0         1.1 MB
    python-3.10.18             |       hc958d9f_0        13.1 MB
    setuptools-78.1.1          |  py310hecd8cb5_0         1.6 MB
    tzdata-2025c               |       he532380_0         118 KB
    wheel-0.45.1               |  py310hecd8cb5_0         119 KB
    ------------------------------------------------------------
 

In [23]:
conda activate pycaret_env


CondaError: Run 'conda init' before 'conda activate'


Note: you may need to restart the kernel to use updated packages.


In [24]:
pip install pycaret

Note: you may need to restart the kernel to use updated packages.


In [25]:
import polars as pl
import numpy as np
import matplotlib.pyplot as plt

from catboost import Pool, CatBoostClassifier
from sklearn.metrics import roc_auc_score

In [26]:
data_path = 'data/'

In [27]:
%%time
#train = pl.read_parquet('data/train_main_features.parquet')
#test = pl.read_parquet('data/test_main_features.parquet')

train = pl.read_parquet(data_path+'train/'+'train_main_features.parquet'
                       ).fill_null(-99)
test = pl.read_parquet(data_path+'test/'+'test_main_features.parquet'
                      ).fill_null(-99)

print('Тренировочные данные:', train.shape)
print('Тестовые данные:', test.shape)
train

Тренировочные данные: (750000, 200)
Тестовые данные: (250000, 200)
CPU times: user 671 ms, sys: 2.07 s, total: 2.74 s
Wall time: 724 ms


customer_id,cat_feature_1,cat_feature_2,cat_feature_3,cat_feature_4,cat_feature_5,cat_feature_6,cat_feature_7,cat_feature_8,cat_feature_9,cat_feature_10,cat_feature_11,cat_feature_12,cat_feature_13,cat_feature_14,cat_feature_15,cat_feature_16,cat_feature_17,cat_feature_18,cat_feature_19,cat_feature_20,cat_feature_21,cat_feature_22,cat_feature_23,cat_feature_24,cat_feature_25,cat_feature_26,cat_feature_27,cat_feature_28,cat_feature_29,cat_feature_30,cat_feature_31,cat_feature_32,cat_feature_33,cat_feature_34,cat_feature_35,cat_feature_36,…,num_feature_96,num_feature_97,num_feature_98,num_feature_99,num_feature_100,num_feature_101,num_feature_102,num_feature_103,num_feature_104,num_feature_105,num_feature_106,num_feature_107,num_feature_108,num_feature_109,num_feature_110,num_feature_111,num_feature_112,num_feature_113,num_feature_114,num_feature_115,num_feature_116,num_feature_117,num_feature_118,num_feature_119,num_feature_120,num_feature_121,num_feature_122,num_feature_123,num_feature_124,num_feature_125,num_feature_126,num_feature_127,num_feature_128,num_feature_129,num_feature_130,num_feature_131,num_feature_132
i32,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,…,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64
1000001,1.0,0.0,2.0,1.0,2.0,3.0,2.0,2.0,4.0,2.0,0.0,0.0,0.0,1.0,2.0,0.0,2.0,2.0,2.0,0.0,0.0,0.0,2.0,0.0,0.0,0.0,1.0,2.0,1.0,2.0,2.0,0.0,0.0,212.0,0.0,0.0,…,-0.284519,-99.0,-0.004499,-99.0,-0.050159,-0.002297,-0.043592,-0.061488,0.450342,-99.0,0.0,-99.0,-0.060492,-0.279105,-99.0,-0.429813,-99.0,-0.009654,-0.293036,-99.0,-0.493959,-0.019079,-99.0,-99.0,-99.0,-99.0,-0.001357,-0.031281,-0.046146,-99.0,-0.445279,-99.0,-99.0,-0.107666,-0.418616,-99.0,-99.0
1000002,1.0,0.0,0.0,1.0,0.0,3.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,1.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,1.0,0.0,212.0,1.0,0.0,…,-0.284519,0.460275,-0.004278,-99.0,-0.556244,-0.002297,-0.043592,-0.061488,1.998252,-0.009552,0.0,-0.005762,-0.060492,-0.279105,-99.0,-0.429813,-99.0,-0.009654,-0.293036,-0.004421,-0.256445,-0.014154,-99.0,-0.24167,-99.0,-99.0,-0.001357,-0.031281,-0.046146,-0.10217,1.550722,-99.0,-99.0,-0.170724,-0.805771,-0.397803,-0.373734
1000003,1.0,0.0,0.0,1.0,0.0,3.0,0.0,0.0,4.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,2.0,0.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,212.0,0.0,0.0,…,-0.284519,-0.264397,-0.004278,-99.0,-99.0,-0.002297,-0.043592,-0.061488,-0.264078,-99.0,0.0,-0.26649,-0.060492,-0.279105,-99.0,-0.429813,-99.0,-0.009654,-0.293036,-0.004421,-0.57313,-0.019124,-99.0,-0.24167,-99.0,-99.0,-0.001357,-0.031281,-0.046146,-99.0,-0.475778,-99.0,-99.0,-0.170724,-0.602005,-0.397803,-0.373734
1000004,1.0,0.0,2.0,1.0,2.0,3.0,2.0,2.0,3.0,2.0,0.0,0.0,0.0,1.0,2.0,2.0,2.0,2.0,2.0,0.0,0.0,2.0,2.0,0.0,0.0,0.0,0.0,2.0,1.0,2.0,2.0,0.0,0.0,212.0,0.0,0.0,…,-0.284519,-99.0,-0.004499,-99.0,-99.0,-0.002297,-0.043592,-0.061488,0.688482,-0.009552,0.0,-99.0,-0.060492,-99.0,-99.0,-0.429813,-99.0,-0.009654,-0.293036,-99.0,-0.57313,-99.0,-99.0,-0.505441,-99.0,0.714631,-0.001357,-0.031281,-0.046146,-99.0,-0.475778,0.111196,0.116695,-99.0,-0.724265,-99.0,-99.0
1000005,1.0,2.0,0.0,1.0,0.0,3.0,0.0,0.0,2.0,1.0,2.0,2.0,2.0,0.0,1.0,0.0,0.0,0.0,0.0,2.0,2.0,2.0,0.0,2.0,2.0,2.0,2.0,0.0,0.0,0.0,0.0,2.0,2.0,212.0,2.0,2.0,…,-0.284519,-0.264397,-0.004499,-99.0,-99.0,-0.002297,-0.043592,-99.0,-0.264078,-99.0,0.0,-0.26649,-99.0,-0.279105,-99.0,-0.429813,-99.0,-0.009654,-0.293036,-0.004421,-0.57313,-0.018674,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-0.046146,-99.0,-99.0,-99.0,-99.0,-0.107666,-99.0,-0.397803,-0.373734
…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…
1749996,1.0,0.0,2.0,0.0,2.0,3.0,2.0,2.0,4.0,2.0,0.0,0.0,0.0,1.0,2.0,0.0,2.0,2.0,2.0,0.0,0.0,2.0,2.0,0.0,0.0

In [28]:
missing_values = (
    train.null_count()
    .unpivot(variable_name="Column", value_name="Missing Values")
    .filter(pl.col("Missing Values") > 0)
    .sort("Missing Values", descending=True)
)

print('\nКоличество пропусков в признаках:\n', missing_values)


Количество пропусков в признаках:
 shape: (0, 2)
┌────────┬────────────────┐
│ Column ┆ Missing Values │
│ ---    ┆ ---            │
│ str    ┆ u32            │
╞════════╪════════════════╡
└────────┴────────────────┘


In [29]:
cat_feature_names = [
    col_name for col_name in train.columns 
    if col_name.startswith("cat_feature")
]
cat_feature_names

['cat_feature_1',
 'cat_feature_2',
 'cat_feature_3',
 'cat_feature_4',
 'cat_feature_5',
 'cat_feature_6',
 'cat_feature_7',
 'cat_feature_8',
 'cat_feature_9',
 'cat_feature_10',
 'cat_feature_11',
 'cat_feature_12',
 'cat_feature_13',
 'cat_feature_14',
 'cat_feature_15',
 'cat_feature_16',
 'cat_feature_17',
 'cat_feature_18',
 'cat_feature_19',
 'cat_feature_20',
 'cat_feature_21',
 'cat_feature_22',
 'cat_feature_23',
 'cat_feature_24',
 'cat_feature_25',
 'cat_feature_26',
 'cat_feature_27',
 'cat_feature_28',
 'cat_feature_29',
 'cat_feature_30',
 'cat_feature_31',
 'cat_feature_32',
 'cat_feature_33',
 'cat_feature_34',
 'cat_feature_35',
 'cat_feature_36',
 'cat_feature_37',
 'cat_feature_38',
 'cat_feature_39',
 'cat_feature_40',
 'cat_feature_41',
 'cat_feature_42',
 'cat_feature_43',
 'cat_feature_44',
 'cat_feature_45',
 'cat_feature_46',
 'cat_feature_47',
 'cat_feature_48',
 'cat_feature_49',
 'cat_feature_50',
 'cat_feature_51',
 'cat_feature_52',
 'cat_feature_53',
 '

In [30]:
train = train.with_columns(
    pl.col(cat_feature_names).cast(pl.Int32)
)
train 

customer_id,cat_feature_1,cat_feature_2,cat_feature_3,cat_feature_4,cat_feature_5,cat_feature_6,cat_feature_7,cat_feature_8,cat_feature_9,cat_feature_10,cat_feature_11,cat_feature_12,cat_feature_13,cat_feature_14,cat_feature_15,cat_feature_16,cat_feature_17,cat_feature_18,cat_feature_19,cat_feature_20,cat_feature_21,cat_feature_22,cat_feature_23,cat_feature_24,cat_feature_25,cat_feature_26,cat_feature_27,cat_feature_28,cat_feature_29,cat_feature_30,cat_feature_31,cat_feature_32,cat_feature_33,cat_feature_34,cat_feature_35,cat_feature_36,…,num_feature_96,num_feature_97,num_feature_98,num_feature_99,num_feature_100,num_feature_101,num_feature_102,num_feature_103,num_feature_104,num_feature_105,num_feature_106,num_feature_107,num_feature_108,num_feature_109,num_feature_110,num_feature_111,num_feature_112,num_feature_113,num_feature_114,num_feature_115,num_feature_116,num_feature_117,num_feature_118,num_feature_119,num_feature_120,num_feature_121,num_feature_122,num_feature_123,num_feature_124,num_feature_125,num_feature_126,num_feature_127,num_feature_128,num_feature_129,num_feature_130,num_feature_131,num_feature_132
i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,…,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64
1000001,1,0,2,1,2,3,2,2,4,2,0,0,0,1,2,0,2,2,2,0,0,0,2,0,0,0,1,2,1,2,2,0,0,212,0,0,…,-0.284519,-99.0,-0.004499,-99.0,-0.050159,-0.002297,-0.043592,-0.061488,0.450342,-99.0,0.0,-99.0,-0.060492,-0.279105,-99.0,-0.429813,-99.0,-0.009654,-0.293036,-99.0,-0.493959,-0.019079,-99.0,-99.0,-99.0,-99.0,-0.001357,-0.031281,-0.046146,-99.0,-0.445279,-99.0,-99.0,-0.107666,-0.418616,-99.0,-99.0
1000002,1,0,0,1,0,3,0,0,0,0,0,0,0,0,1,0,0,0,0,1,0,0,0,1,0,0,1,0,0,0,0,1,0,212,1,0,…,-0.284519,0.460275,-0.004278,-99.0,-0.556244,-0.002297,-0.043592,-0.061488,1.998252,-0.009552,0.0,-0.005762,-0.060492,-0.279105,-99.0,-0.429813,-99.0,-0.009654,-0.293036,-0.004421,-0.256445,-0.014154,-99.0,-0.24167,-99.0,-99.0,-0.001357,-0.031281,-0.046146,-0.10217,1.550722,-99.0,-99.0,-0.170724,-0.805771,-0.397803,-0.373734
1000003,1,0,0,1,0,3,0,0,4,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,1,0,0,0,0,0,0,212,0,0,…,-0.284519,-0.264397,-0.004278,-99.0,-99.0,-0.002297,-0.043592,-0.061488,-0.264078,-99.0,0.0,-0.26649,-0.060492,-0.279105,-99.0,-0.429813,-99.0,-0.009654,-0.293036,-0.004421,-0.57313,-0.019124,-99.0,-0.24167,-99.0,-99.0,-0.001357,-0.031281,-0.046146,-99.0,-0.475778,-99.0,-99.0,-0.170724,-0.602005,-0.397803,-0.373734
1000004,1,0,2,1,2,3,2,2,3,2,0,0,0,1,2,2,2,2,2,0,0,2,2,0,0,0,0,2,1,2,2,0,0,212,0,0,…,-0.284519,-99.0,-0.004499,-99.0,-99.0,-0.002297,-0.043592,-0.061488,0.688482,-0.009552,0.0,-99.0,-0.060492,-99.0,-99.0,-0.429813,-99.0,-0.009654,-0.293036,-99.0,-0.57313,-99.0,-99.0,-0.505441,-99.0,0.714631,-0.001357,-0.031281,-0.046146,-99.0,-0.475778,0.111196,0.116695,-99.0,-0.724265,-99.0,-99.0
1000005,1,2,0,1,0,3,0,0,2,1,2,2,2,0,1,0,0,0,0,2,2,2,0,2,2,2,2,0,0,0,0,2,2,212,2,2,…,-0.284519,-0.264397,-0.004499,-99.0,-99.0,-0.002297,-0.043592,-99.0,-0.264078,-99.0,0.0,-0.26649,-99.0,-0.279105,-99.0,-0.429813,-99.0,-0.009654,-0.293036,-0.004421,-0.57313,-0.018674,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-0.046146,-99.0,-99.0,-99.0,-99.0,-0.107666,-99.0,-0.397803,-0.373734
…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…
1749996,1,0,2,0,2,3,2,2,4,2,0,0,0,1,2,0,2,2,2,0,0,2,2,0,0,0,0,2,1,2,2,0,0,212,0,0,…,-99.0,-99.0,-0.004499,-0.150243,-99.0,-99.0,-99.0,2.831928,-99.0,-99.0,-99.0,-99.0,-0.060492,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-0.019236,-99.0,-0.505441,-99.0,-99.0,-0.001357,-0.031281,-99.0,-99.0,-0.475778,-99.0,-99.0,-99.0,0.335315,-99.0,-99.0
1749997,1,0,2,1,2,3,2,2,2,2,0,0,0,1,2,0,2,2,2,0,0,0,2,0,0,0,0,2,1,2,2,0,0,212,1,0,…,-0.284519,-99.0,-0.004499,-99.0,-0.5100

In [31]:
target = pl.read_parquet(data_path+'baseline/'+'train_target.parquet')
target
#shape: (750_000, 42)

customer_id,target_1_1,target_1_2,target_1_3,target_1_4,target_1_5,target_2_1,target_2_2,target_2_3,target_2_4,target_2_5,target_2_6,target_2_7,target_2_8,target_3_1,target_3_2,target_3_3,target_3_4,target_3_5,target_4_1,target_5_1,target_5_2,target_6_1,target_6_2,target_6_3,target_6_4,target_6_5,target_7_1,target_7_2,target_7_3,target_8_1,target_8_2,target_8_3,target_9_1,target_9_2,target_9_3,target_9_4,target_9_5,target_9_6,target_9_7,target_9_8,target_10_1
i32,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64
1000001,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0
1000002,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
1000003,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,1.0
1000004,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0
1000005,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,1.0
…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…
1749996,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0
1749997,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,1.0,0.0,0.0
1749998,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,1.0


In [32]:
target_columns = [col for col in target.columns if col.startswith("target")]
target_columns

['target_1_1',
 'target_1_2',
 'target_1_3',
 'target_1_4',
 'target_1_5',
 'target_2_1',
 'target_2_2',
 'target_2_3',
 'target_2_4',
 'target_2_5',
 'target_2_6',
 'target_2_7',
 'target_2_8',
 'target_3_1',
 'target_3_2',
 'target_3_3',
 'target_3_4',
 'target_3_5',
 'target_4_1',
 'target_5_1',
 'target_5_2',
 'target_6_1',
 'target_6_2',
 'target_6_3',
 'target_6_4',
 'target_6_5',
 'target_7_1',
 'target_7_2',
 'target_7_3',
 'target_8_1',
 'target_8_2',
 'target_8_3',
 'target_9_1',
 'target_9_2',
 'target_9_3',
 'target_9_4',
 'target_9_5',
 'target_9_6',
 'target_9_7',
 'target_9_8',
 'target_10_1']

## Обучение модели
Обучение 41 независимой модели бинарной классификации

In [36]:
model = CatBoostClassifier(
    #one_hot_max_size=1024*2,
    one_hot_max_size=256,
    
    loss_function = 'Logloss', 
    custom_metric=["Accuracy","AUC"],
    eval_metric="AUC",
    nan_mode = 'Min', 
    random_seed = 1234,

    # task_type="GPU",
    task_type="CPU",   # На Mac работает только CPU
    verbose = 100
)

In [37]:
%%time
# limit=500_000
limit=700_000
for tcols in target_columns:
    print(tcols)
    train_pool = Pool(data = train[:limit].drop("customer_id").to_pandas(), 
                      label = (
                          target[:limit]
                              .select(tcols)
                              # .drop("customer_id")
                              .to_pandas(), 
                          # cat_features = cat_feature_names
                      )
                     )

    valid_pool = Pool(data = train[limit:].drop("customer_id").to_pandas(), 
                      label = (
                          target[limit:]
                              .select(tcols)
                              # .drop("customer_id")
                              .to_pandas(),
                          # cat_features = cat_feature_names
                      )
                     )
    
    model.fit(train_pool,
              plot=True,
              eval_set=valid_pool,
             )
    model.save_model(f"wocat{tcols}.cbm")

target_1_1


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.5834810	best: 0.5834810 (0)	total: 167ms	remaining: 2m 46s
100:	test: 0.9067562	best: 0.9067912 (97)	total: 7.32s	remaining: 1m 5s
200:	test: 0.9116913	best: 0.9117064 (199)	total: 13.5s	remaining: 53.5s
300:	test: 0.9145190	best: 0.9145190 (300)	total: 19.3s	remaining: 44.8s
400:	test: 0.9157701	best: 0.9157711 (399)	total: 25s	remaining: 37.4s
500:	test: 0.9167966	best: 0.9168130 (492)	total: 30.8s	remaining: 30.7s
600:	test: 0.9167192	best: 0.9169169 (591)	total: 36.5s	remaining: 24.2s
700:	test: 0.9170815	best: 0.9170910 (697)	total: 42.1s	remaining: 18s
800:	test: 0.9171724	best: 0.9172335 (722)	total: 47.6s	remaining: 11.8s
900:	test: 0.9169574	best: 0.9173695 (808)	total: 53.4s	remaining: 5.87s
999:	test: 0.9169469	best: 0.9173695 (808)	total: 58.9s	remaining: 0us

bestTest = 0.9173694636
bestIteration = 808

Shrink model to first 809 iterations.
target_1_2


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.6331434	best: 0.6331434 (0)	total: 90.6ms	remaining: 1m 30s
100:	test: 0.8029972	best: 0.8034136 (96)	total: 6.19s	remaining: 55.1s
200:	test: 0.8066138	best: 0.8067777 (172)	total: 11.8s	remaining: 46.9s
300:	test: 0.8054688	best: 0.8072139 (205)	total: 17.5s	remaining: 40.7s
400:	test: 0.8040988	best: 0.8072139 (205)	total: 23.1s	remaining: 34.6s
500:	test: 0.8036074	best: 0.8072139 (205)	total: 28.8s	remaining: 28.7s
600:	test: 0.8044930	best: 0.8072139 (205)	total: 34.8s	remaining: 23.1s
700:	test: 0.8048938	best: 0.8072139 (205)	total: 40.7s	remaining: 17.4s
800:	test: 0.8034653	best: 0.8072139 (205)	total: 46.6s	remaining: 11.6s
900:	test: 0.8002411	best: 0.8072139 (205)	total: 52.4s	remaining: 5.76s
999:	test: 0.8008910	best: 0.8072139 (205)	total: 58s	remaining: 0us

bestTest = 0.8072138771
bestIteration = 205

Shrink model to first 206 iterations.
target_1_3


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.7394315	best: 0.7394315 (0)	total: 96.7ms	remaining: 1m 36s
100:	test: 0.8588080	best: 0.8588080 (100)	total: 7.2s	remaining: 1m 4s
200:	test: 0.8622855	best: 0.8622855 (200)	total: 13.7s	remaining: 54.4s
300:	test: 0.8644360	best: 0.8644360 (300)	total: 19.8s	remaining: 46.1s
400:	test: 0.8655849	best: 0.8655849 (400)	total: 25.9s	remaining: 38.7s
500:	test: 0.8662480	best: 0.8662480 (500)	total: 31.8s	remaining: 31.6s
600:	test: 0.8666454	best: 0.8668581 (554)	total: 37.7s	remaining: 25s
700:	test: 0.8669149	best: 0.8669149 (700)	total: 43.5s	remaining: 18.6s
800:	test: 0.8671596	best: 0.8671599 (709)	total: 49.3s	remaining: 12.3s
900:	test: 0.8676250	best: 0.8677109 (886)	total: 55.3s	remaining: 6.07s
999:	test: 0.8678778	best: 0.8678778 (999)	total: 1m 1s	remaining: 0us

bestTest = 0.8678777912
bestIteration = 999

target_1_4


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.6698358	best: 0.6698358 (0)	total: 95.7ms	remaining: 1m 35s
100:	test: 0.8167624	best: 0.8167624 (100)	total: 6.98s	remaining: 1m 2s
200:	test: 0.8198961	best: 0.8199311 (198)	total: 12.9s	remaining: 51.2s
300:	test: 0.8220200	best: 0.8221760 (295)	total: 18.8s	remaining: 43.6s
400:	test: 0.8233756	best: 0.8234507 (392)	total: 24.6s	remaining: 36.7s
500:	test: 0.8237576	best: 0.8238864 (469)	total: 30.5s	remaining: 30.4s
600:	test: 0.8250268	best: 0.8251348 (573)	total: 36.3s	remaining: 24.1s
700:	test: 0.8253163	best: 0.8253235 (696)	total: 41.9s	remaining: 17.9s
800:	test: 0.8244749	best: 0.8256038 (723)	total: 47.8s	remaining: 11.9s
900:	test: 0.8244381	best: 0.8256038 (723)	total: 53.9s	remaining: 5.92s
999:	test: 0.8240684	best: 0.8256038 (723)	total: 59.7s	remaining: 0us

bestTest = 0.8256038076
bestIteration = 723

Shrink model to first 724 iterations.
target_1_5


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.7049976	best: 0.7049976 (0)	total: 86ms	remaining: 1m 25s
100:	test: 0.8484840	best: 0.8486538 (96)	total: 6.15s	remaining: 54.7s
200:	test: 0.8638049	best: 0.8640627 (199)	total: 11.8s	remaining: 46.9s
300:	test: 0.8639601	best: 0.8640627 (199)	total: 17.5s	remaining: 40.6s
400:	test: 0.8647852	best: 0.8649263 (330)	total: 23.3s	remaining: 34.8s
500:	test: 0.8658894	best: 0.8667496 (470)	total: 29.1s	remaining: 29s
600:	test: 0.8642740	best: 0.8667496 (470)	total: 34.7s	remaining: 23s
700:	test: 0.8646598	best: 0.8667496 (470)	total: 40.5s	remaining: 17.3s
800:	test: 0.8672797	best: 0.8675064 (799)	total: 46.2s	remaining: 11.5s
900:	test: 0.8626970	best: 0.8675064 (799)	total: 52.1s	remaining: 5.73s
999:	test: 0.8604127	best: 0.8675064 (799)	total: 58s	remaining: 0us

bestTest = 0.8675064351
bestIteration = 799

Shrink model to first 800 iterations.
target_2_1


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.6074912	best: 0.6074912 (0)	total: 87.5ms	remaining: 1m 27s
100:	test: 0.7919764	best: 0.7919764 (100)	total: 6.48s	remaining: 57.7s
200:	test: 0.7924678	best: 0.7932700 (128)	total: 12s	remaining: 47.8s
300:	test: 0.7949030	best: 0.7949030 (300)	total: 17.6s	remaining: 40.8s
400:	test: 0.7943884	best: 0.7953405 (378)	total: 23.1s	remaining: 34.5s
500:	test: 0.7940836	best: 0.7953405 (378)	total: 28.7s	remaining: 28.5s
600:	test: 0.7927316	best: 0.7953405 (378)	total: 34.3s	remaining: 22.8s
700:	test: 0.7922071	best: 0.7953405 (378)	total: 40s	remaining: 17.1s
800:	test: 0.7910999	best: 0.7953405 (378)	total: 45.5s	remaining: 11.3s
900:	test: 0.7877982	best: 0.7953405 (378)	total: 51s	remaining: 5.61s
999:	test: 0.7865807	best: 0.7953405 (378)	total: 56.7s	remaining: 0us

bestTest = 0.7953404891
bestIteration = 378

Shrink model to first 379 iterations.
target_2_2


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.7462089	best: 0.7462089 (0)	total: 105ms	remaining: 1m 44s
100:	test: 0.9145183	best: 0.9145183 (100)	total: 6.82s	remaining: 1m
200:	test: 0.9175598	best: 0.9175688 (199)	total: 13.1s	remaining: 51.9s
300:	test: 0.9187040	best: 0.9187102 (299)	total: 19s	remaining: 44.1s
400:	test: 0.9198585	best: 0.9198640 (399)	total: 24.9s	remaining: 37.2s
500:	test: 0.9207927	best: 0.9207927 (500)	total: 31.5s	remaining: 31.3s
600:	test: 0.9208974	best: 0.9210608 (527)	total: 37.7s	remaining: 25s
700:	test: 0.9215093	best: 0.9215210 (696)	total: 44.2s	remaining: 18.9s
800:	test: 0.9218603	best: 0.9218603 (800)	total: 50.4s	remaining: 12.5s
900:	test: 0.9222211	best: 0.9222323 (899)	total: 56.5s	remaining: 6.21s
999:	test: 0.9222233	best: 0.9223168 (941)	total: 1m 2s	remaining: 0us

bestTest = 0.9223167817
bestIteration = 941

Shrink model to first 942 iterations.
target_2_3


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.6189671	best: 0.6189671 (0)	total: 94.8ms	remaining: 1m 34s
100:	test: 0.8051981	best: 0.8052939 (96)	total: 5.62s	remaining: 50s
200:	test: 0.8017095	best: 0.8057130 (101)	total: 11s	remaining: 43.7s
300:	test: 0.7996601	best: 0.8057130 (101)	total: 16.4s	remaining: 38s
400:	test: 0.7900041	best: 0.8057130 (101)	total: 21.7s	remaining: 32.4s
500:	test: 0.7868383	best: 0.8057130 (101)	total: 27s	remaining: 26.8s
600:	test: 0.7840092	best: 0.8057130 (101)	total: 32.5s	remaining: 21.6s
700:	test: 0.7815536	best: 0.8057130 (101)	total: 38.1s	remaining: 16.2s
800:	test: 0.7724085	best: 0.8057130 (101)	total: 43.4s	remaining: 10.8s
900:	test: 0.7680747	best: 0.8057130 (101)	total: 48.8s	remaining: 5.36s
999:	test: 0.7573448	best: 0.8057130 (101)	total: 54.3s	remaining: 0us

bestTest = 0.8057130434
bestIteration = 101

Shrink model to first 102 iterations.
target_2_4


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.5356780	best: 0.5356780 (0)	total: 82.6ms	remaining: 1m 22s
100:	test: 0.7262570	best: 0.7262570 (100)	total: 6.17s	remaining: 54.9s
200:	test: 0.7258951	best: 0.7281150 (139)	total: 11.9s	remaining: 47.3s
300:	test: 0.7245361	best: 0.7281150 (139)	total: 17.6s	remaining: 41s
400:	test: 0.7233854	best: 0.7281150 (139)	total: 23.6s	remaining: 35.3s
500:	test: 0.7242089	best: 0.7281150 (139)	total: 29.4s	remaining: 29.3s
600:	test: 0.7235093	best: 0.7281150 (139)	total: 35.1s	remaining: 23.3s
700:	test: 0.7197599	best: 0.7281150 (139)	total: 40.6s	remaining: 17.3s
800:	test: 0.7193921	best: 0.7281150 (139)	total: 46.4s	remaining: 11.5s
900:	test: 0.7199816	best: 0.7281150 (139)	total: 52s	remaining: 5.72s
999:	test: 0.7198469	best: 0.7281150 (139)	total: 57.7s	remaining: 0us

bestTest = 0.7281150335
bestIteration = 139

Shrink model to first 140 iterations.
target_2_5


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.5697830	best: 0.5697830 (0)	total: 68.4ms	remaining: 1m 8s
100:	test: 0.7502219	best: 0.7509310 (90)	total: 5.84s	remaining: 52s
200:	test: 0.7632422	best: 0.7632422 (200)	total: 11.1s	remaining: 44.2s
300:	test: 0.7680823	best: 0.7684746 (297)	total: 16.7s	remaining: 38.8s
400:	test: 0.7699380	best: 0.7699380 (400)	total: 22.1s	remaining: 33s
500:	test: 0.7729802	best: 0.7761488 (456)	total: 28.2s	remaining: 28.1s
600:	test: 0.7718461	best: 0.7761488 (456)	total: 33.9s	remaining: 22.5s
700:	test: 0.7759041	best: 0.7761517 (696)	total: 39.4s	remaining: 16.8s
800:	test: 0.7727379	best: 0.7761517 (696)	total: 45.1s	remaining: 11.2s
900:	test: 0.7678807	best: 0.7761517 (696)	total: 50.6s	remaining: 5.56s
999:	test: 0.7680461	best: 0.7761517 (696)	total: 56.1s	remaining: 0us

bestTest = 0.7761517409
bestIteration = 696

Shrink model to first 697 iterations.
target_2_6


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.5092151	best: 0.5092151 (0)	total: 78.6ms	remaining: 1m 18s
100:	test: 0.7192149	best: 0.7194900 (98)	total: 6.08s	remaining: 54.1s
200:	test: 0.7299062	best: 0.7299062 (200)	total: 11.6s	remaining: 46.2s
300:	test: 0.7339381	best: 0.7352296 (291)	total: 17.1s	remaining: 39.7s
400:	test: 0.7373953	best: 0.7385995 (392)	total: 22.6s	remaining: 33.7s
500:	test: 0.7392433	best: 0.7394049 (497)	total: 27.9s	remaining: 27.8s
600:	test: 0.7398487	best: 0.7401689 (532)	total: 33.7s	remaining: 22.4s
700:	test: 0.7389013	best: 0.7405905 (607)	total: 39.2s	remaining: 16.7s
800:	test: 0.7398740	best: 0.7405905 (607)	total: 44.7s	remaining: 11.1s
900:	test: 0.7387609	best: 0.7405905 (607)	total: 50.3s	remaining: 5.53s
999:	test: 0.7321877	best: 0.7405905 (607)	total: 55.9s	remaining: 0us

bestTest = 0.7405905141
bestIteration = 607

Shrink model to first 608 iterations.
target_2_7


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.7710693	best: 0.7710693 (0)	total: 85.1ms	remaining: 1m 25s
100:	test: 0.9291654	best: 0.9299070 (92)	total: 5.66s	remaining: 50.4s
200:	test: 0.9349071	best: 0.9360662 (195)	total: 10.9s	remaining: 43.5s
300:	test: 0.9311580	best: 0.9363622 (221)	total: 16.2s	remaining: 37.6s
400:	test: 0.9352499	best: 0.9397873 (362)	total: 21.5s	remaining: 32.2s
500:	test: 0.9276103	best: 0.9397873 (362)	total: 27.1s	remaining: 27s
600:	test: 0.9220219	best: 0.9397873 (362)	total: 32.4s	remaining: 21.5s
700:	test: 0.9198706	best: 0.9397873 (362)	total: 37.9s	remaining: 16.2s
800:	test: 0.9220353	best: 0.9397873 (362)	total: 43.4s	remaining: 10.8s
900:	test: 0.9134354	best: 0.9397873 (362)	total: 48.7s	remaining: 5.35s
999:	test: 0.9088460	best: 0.9397873 (362)	total: 54.2s	remaining: 0us

bestTest = 0.9397872695
bestIteration = 362

Shrink model to first 363 iterations.
target_2_8


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.8374870	best: 0.8374870 (0)	total: 88.2ms	remaining: 1m 28s
100:	test: 0.9437205	best: 0.9757206 (25)	total: 6.21s	remaining: 55.3s
200:	test: 0.9396652	best: 0.9757206 (25)	total: 11.6s	remaining: 46s
300:	test: 0.9168633	best: 0.9757206 (25)	total: 17.1s	remaining: 39.7s
400:	test: 0.9067275	best: 0.9757206 (25)	total: 22.8s	remaining: 34s
500:	test: 0.9017571	best: 0.9757206 (25)	total: 28.4s	remaining: 28.3s
600:	test: 0.9033173	best: 0.9757206 (25)	total: 34s	remaining: 22.6s
700:	test: 0.8902012	best: 0.9757206 (25)	total: 39.7s	remaining: 16.9s
800:	test: 0.8797004	best: 0.9757206 (25)	total: 45.2s	remaining: 11.2s
900:	test: 0.8763501	best: 0.9757206 (25)	total: 50.8s	remaining: 5.59s
999:	test: 0.8801554	best: 0.9757206 (25)	total: 56.2s	remaining: 0us

bestTest = 0.9757205576
bestIteration = 25

Shrink model to first 26 iterations.
target_3_1


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.5783038	best: 0.5783038 (0)	total: 88ms	remaining: 1m 27s
100:	test: 0.6762041	best: 0.6762041 (100)	total: 7.02s	remaining: 1m 2s
200:	test: 0.6814380	best: 0.6814380 (200)	total: 13.3s	remaining: 52.8s
300:	test: 0.6835676	best: 0.6835676 (300)	total: 19.5s	remaining: 45.3s
400:	test: 0.6856011	best: 0.6856800 (394)	total: 25.6s	remaining: 38.3s
500:	test: 0.6859508	best: 0.6861739 (484)	total: 31.8s	remaining: 31.6s
600:	test: 0.6863508	best: 0.6864030 (599)	total: 37.7s	remaining: 25s
700:	test: 0.6864466	best: 0.6865220 (692)	total: 43.8s	remaining: 18.7s
800:	test: 0.6866491	best: 0.6867044 (722)	total: 49.9s	remaining: 12.4s
900:	test: 0.6870640	best: 0.6872747 (884)	total: 55.9s	remaining: 6.14s
999:	test: 0.6865046	best: 0.6872747 (884)	total: 1m 1s	remaining: 0us

bestTest = 0.6872747331
bestIteration = 884

Shrink model to first 885 iterations.
target_3_2


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.8477778	best: 0.8477778 (0)	total: 85.4ms	remaining: 1m 25s
100:	test: 0.9019667	best: 0.9019667 (100)	total: 6.73s	remaining: 59.9s
200:	test: 0.9040772	best: 0.9040772 (200)	total: 12.9s	remaining: 51.3s
300:	test: 0.9050948	best: 0.9050987 (299)	total: 19.1s	remaining: 44.3s
400:	test: 0.9054082	best: 0.9054934 (390)	total: 25.2s	remaining: 37.6s
500:	test: 0.9056965	best: 0.9056965 (500)	total: 31.4s	remaining: 31.3s
600:	test: 0.9059050	best: 0.9059050 (600)	total: 37.5s	remaining: 24.9s
700:	test: 0.9060268	best: 0.9060821 (689)	total: 43.9s	remaining: 18.7s
800:	test: 0.9061688	best: 0.9061695 (798)	total: 50.5s	remaining: 12.5s
900:	test: 0.9063349	best: 0.9063349 (900)	total: 56.8s	remaining: 6.24s
999:	test: 0.9061555	best: 0.9063569 (911)	total: 1m 2s	remaining: 0us

bestTest = 0.9063568756
bestIteration = 911

Shrink model to first 912 iterations.
target_3_3


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.5022632	best: 0.5022632 (0)	total: 92ms	remaining: 1m 31s
100:	test: 0.7761875	best: 0.7761875 (100)	total: 5.95s	remaining: 53s
200:	test: 0.7805389	best: 0.7816245 (126)	total: 11.9s	remaining: 47.4s
300:	test: 0.7693242	best: 0.7835473 (205)	total: 17.3s	remaining: 40.2s
400:	test: 0.7590108	best: 0.7835473 (205)	total: 22.7s	remaining: 34s
500:	test: 0.7482177	best: 0.7835473 (205)	total: 28.2s	remaining: 28.1s
600:	test: 0.7431077	best: 0.7835473 (205)	total: 33.5s	remaining: 22.3s
700:	test: 0.7310633	best: 0.7835473 (205)	total: 39s	remaining: 16.6s
800:	test: 0.7301148	best: 0.7835473 (205)	total: 44.4s	remaining: 11s
900:	test: 0.7294126	best: 0.7835473 (205)	total: 50s	remaining: 5.49s
999:	test: 0.7277538	best: 0.7835473 (205)	total: 55.3s	remaining: 0us

bestTest = 0.7835473422
bestIteration = 205

Shrink model to first 206 iterations.
target_3_4


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.5727312	best: 0.5727312 (0)	total: 91.9ms	remaining: 1m 31s
100:	test: 0.9328333	best: 0.9360063 (62)	total: 5.87s	remaining: 52.3s
200:	test: 0.9338279	best: 0.9360063 (62)	total: 11.3s	remaining: 44.9s
300:	test: 0.9354642	best: 0.9360063 (62)	total: 16.7s	remaining: 38.7s
400:	test: 0.9356560	best: 0.9360313 (361)	total: 22.2s	remaining: 33.2s
500:	test: 0.9371537	best: 0.9372570 (475)	total: 27.6s	remaining: 27.5s
600:	test: 0.9362336	best: 0.9375232 (549)	total: 33.1s	remaining: 22s
700:	test: 0.9363687	best: 0.9375232 (549)	total: 38.6s	remaining: 16.5s
800:	test: 0.9374921	best: 0.9384468 (778)	total: 44.2s	remaining: 11s
900:	test: 0.9364522	best: 0.9384468 (778)	total: 49.7s	remaining: 5.46s
999:	test: 0.9350371	best: 0.9384468 (778)	total: 55.2s	remaining: 0us

bestTest = 0.9384468179
bestIteration = 778

Shrink model to first 779 iterations.
target_3_5


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.6021122	best: 0.6021122 (0)	total: 95ms	remaining: 1m 34s
100:	test: 0.9692405	best: 0.9692405 (100)	total: 5.73s	remaining: 51s
200:	test: 0.9697370	best: 0.9701291 (184)	total: 11.2s	remaining: 44.4s
300:	test: 0.9717765	best: 0.9720586 (294)	total: 16.8s	remaining: 39s
400:	test: 0.9747137	best: 0.9747884 (391)	total: 22.5s	remaining: 33.6s
500:	test: 0.9741715	best: 0.9747884 (391)	total: 27.9s	remaining: 27.8s
600:	test: 0.9750530	best: 0.9751690 (588)	total: 33.4s	remaining: 22.2s
700:	test: 0.9762525	best: 0.9762525 (700)	total: 39.3s	remaining: 16.7s
800:	test: 0.9759168	best: 0.9763492 (786)	total: 44.8s	remaining: 11.1s
900:	test: 0.9760409	best: 0.9763873 (882)	total: 50.5s	remaining: 5.54s
999:	test: 0.9761365	best: 0.9767064 (948)	total: 56.1s	remaining: 0us

bestTest = 0.9767063597
bestIteration = 948

Shrink model to first 949 iterations.
target_4_1


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.5071855	best: 0.5071855 (0)	total: 96.7ms	remaining: 1m 36s
100:	test: 0.8486621	best: 0.8486621 (100)	total: 6.2s	remaining: 55.2s
200:	test: 0.8523848	best: 0.8523894 (199)	total: 11.7s	remaining: 46.5s
300:	test: 0.8539860	best: 0.8540357 (298)	total: 17.3s	remaining: 40.1s
400:	test: 0.8558030	best: 0.8558335 (386)	total: 23s	remaining: 34.4s
500:	test: 0.8569962	best: 0.8570499 (474)	total: 29.1s	remaining: 29s
600:	test: 0.8559466	best: 0.8571825 (504)	total: 34.9s	remaining: 23.2s
700:	test: 0.8557473	best: 0.8571825 (504)	total: 40.5s	remaining: 17.3s
800:	test: 0.8553017	best: 0.8571825 (504)	total: 46.1s	remaining: 11.5s
900:	test: 0.8551311	best: 0.8571825 (504)	total: 51.9s	remaining: 5.7s
999:	test: 0.8541348	best: 0.8571825 (504)	total: 57.7s	remaining: 0us

bestTest = 0.8571825304
bestIteration = 504

Shrink model to first 505 iterations.
target_5_1


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.5043140	best: 0.5043140 (0)	total: 88.1ms	remaining: 1m 28s
100:	test: 0.7349130	best: 0.7349154 (99)	total: 5.99s	remaining: 53.4s
200:	test: 0.7357962	best: 0.7362650 (175)	total: 11.3s	remaining: 45.1s
300:	test: 0.7374139	best: 0.7379318 (283)	total: 16.7s	remaining: 38.7s
400:	test: 0.7367073	best: 0.7379318 (283)	total: 22.2s	remaining: 33.2s
500:	test: 0.7367305	best: 0.7379318 (283)	total: 27.6s	remaining: 27.5s
600:	test: 0.7356296	best: 0.7379318 (283)	total: 33.1s	remaining: 22s
700:	test: 0.7350448	best: 0.7379318 (283)	total: 38.5s	remaining: 16.4s
800:	test: 0.7348582	best: 0.7379318 (283)	total: 44.3s	remaining: 11s
900:	test: 0.7309012	best: 0.7379318 (283)	total: 50s	remaining: 5.49s
999:	test: 0.7309381	best: 0.7379318 (283)	total: 55.5s	remaining: 0us

bestTest = 0.7379318288
bestIteration = 283

Shrink model to first 284 iterations.
target_5_2


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.4853230	best: 0.4853230 (0)	total: 94.3ms	remaining: 1m 34s
100:	test: 0.6985722	best: 0.6996773 (86)	total: 5.96s	remaining: 53s
200:	test: 0.6997652	best: 0.7047915 (166)	total: 11.4s	remaining: 45.4s
300:	test: 0.7034071	best: 0.7051580 (292)	total: 16.6s	remaining: 38.7s
400:	test: 0.7046751	best: 0.7051580 (292)	total: 22.1s	remaining: 33s
500:	test: 0.7085253	best: 0.7085253 (500)	total: 27.4s	remaining: 27.3s
600:	test: 0.7116726	best: 0.7128707 (593)	total: 32.8s	remaining: 21.7s
700:	test: 0.7092740	best: 0.7128707 (593)	total: 38.2s	remaining: 16.3s
800:	test: 0.7046228	best: 0.7128707 (593)	total: 43.9s	remaining: 10.9s
900:	test: 0.7030288	best: 0.7128707 (593)	total: 49.4s	remaining: 5.43s
999:	test: 0.6975567	best: 0.7128707 (593)	total: 54.8s	remaining: 0us

bestTest = 0.7128707199
bestIteration = 593

Shrink model to first 594 iterations.
target_6_1


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.5181069	best: 0.5181069 (0)	total: 98.5ms	remaining: 1m 38s
100:	test: 0.6922229	best: 0.6925588 (99)	total: 5.94s	remaining: 52.9s
200:	test: 0.6945541	best: 0.6947401 (197)	total: 11.4s	remaining: 45.3s
300:	test: 0.6948024	best: 0.6951960 (234)	total: 16.9s	remaining: 39.2s
400:	test: 0.6926610	best: 0.6951960 (234)	total: 22.4s	remaining: 33.5s
500:	test: 0.6954552	best: 0.6954591 (499)	total: 28s	remaining: 27.9s
600:	test: 0.6941804	best: 0.6956025 (507)	total: 33.7s	remaining: 22.3s
700:	test: 0.6955898	best: 0.6960813 (692)	total: 39.2s	remaining: 16.7s
800:	test: 0.6933919	best: 0.6960813 (692)	total: 44.7s	remaining: 11.1s
900:	test: 0.6912209	best: 0.6960813 (692)	total: 50.5s	remaining: 5.55s
999:	test: 0.6893549	best: 0.6960813 (692)	total: 56.2s	remaining: 0us

bestTest = 0.6960812554
bestIteration = 692

Shrink model to first 693 iterations.
target_6_2


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.5564189	best: 0.5564189 (0)	total: 91.5ms	remaining: 1m 31s
100:	test: 0.7128788	best: 0.7129224 (98)	total: 6.15s	remaining: 54.8s
200:	test: 0.7180407	best: 0.7182576 (198)	total: 11.8s	remaining: 47s
300:	test: 0.7217568	best: 0.7217568 (300)	total: 17.2s	remaining: 40.1s
400:	test: 0.7205922	best: 0.7223312 (326)	total: 22.7s	remaining: 33.9s
500:	test: 0.7208235	best: 0.7223312 (326)	total: 28s	remaining: 27.9s
600:	test: 0.7199895	best: 0.7223312 (326)	total: 33.6s	remaining: 22.3s
700:	test: 0.7198447	best: 0.7223312 (326)	total: 39.1s	remaining: 16.7s
800:	test: 0.7217659	best: 0.7223312 (326)	total: 44.7s	remaining: 11.1s
900:	test: 0.7232098	best: 0.7232098 (900)	total: 50.2s	remaining: 5.52s
999:	test: 0.7229628	best: 0.7233722 (945)	total: 55.9s	remaining: 0us

bestTest = 0.7233721752
bestIteration = 945

Shrink model to first 946 iterations.
target_6_3


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.4916201	best: 0.4916201 (0)	total: 79.4ms	remaining: 1m 19s
100:	test: 0.7500473	best: 0.7500666 (99)	total: 5.97s	remaining: 53.2s
200:	test: 0.7517285	best: 0.7526051 (142)	total: 11.5s	remaining: 45.6s
300:	test: 0.7586773	best: 0.7586773 (300)	total: 16.9s	remaining: 39.3s
400:	test: 0.7592171	best: 0.7606242 (378)	total: 22.4s	remaining: 33.4s
500:	test: 0.7577464	best: 0.7606242 (378)	total: 27.7s	remaining: 27.6s
600:	test: 0.7547847	best: 0.7606242 (378)	total: 33.2s	remaining: 22s
700:	test: 0.7542866	best: 0.7606242 (378)	total: 38.6s	remaining: 16.5s
800:	test: 0.7551076	best: 0.7606242 (378)	total: 44.3s	remaining: 11s
900:	test: 0.7549023	best: 0.7606242 (378)	total: 50.4s	remaining: 5.53s
999:	test: 0.7528089	best: 0.7606242 (378)	total: 56.3s	remaining: 0us

bestTest = 0.7606242296
bestIteration = 378

Shrink model to first 379 iterations.
target_6_4


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.5458916	best: 0.5458916 (0)	total: 93ms	remaining: 1m 32s
100:	test: 0.8477129	best: 0.8481474 (95)	total: 6.2s	remaining: 55.2s
200:	test: 0.8502172	best: 0.8513576 (157)	total: 11.8s	remaining: 47s
300:	test: 0.8501080	best: 0.8513576 (157)	total: 17.5s	remaining: 40.6s
400:	test: 0.8501967	best: 0.8513576 (157)	total: 23.1s	remaining: 34.5s
500:	test: 0.8503669	best: 0.8513576 (157)	total: 28.7s	remaining: 28.6s
600:	test: 0.8481929	best: 0.8513576 (157)	total: 34.5s	remaining: 22.9s
700:	test: 0.8488356	best: 0.8513576 (157)	total: 40.3s	remaining: 17.2s
800:	test: 0.8472532	best: 0.8513576 (157)	total: 46s	remaining: 11.4s
900:	test: 0.8471494	best: 0.8513576 (157)	total: 52s	remaining: 5.72s
999:	test: 0.8468949	best: 0.8513576 (157)	total: 58.5s	remaining: 0us

bestTest = 0.8513576094
bestIteration = 157

Shrink model to first 158 iterations.
target_6_5


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.5750346	best: 0.5750346 (0)	total: 99.1ms	remaining: 1m 39s
100:	test: 0.8506092	best: 0.8525659 (65)	total: 5.97s	remaining: 53.2s
200:	test: 0.8494506	best: 0.8525659 (65)	total: 11.3s	remaining: 45s
300:	test: 0.8556521	best: 0.8563484 (252)	total: 16.8s	remaining: 38.9s
400:	test: 0.8554472	best: 0.8608986 (376)	total: 22s	remaining: 32.9s
500:	test: 0.8550118	best: 0.8608986 (376)	total: 27.3s	remaining: 27.2s
600:	test: 0.8548979	best: 0.8608986 (376)	total: 32.6s	remaining: 21.6s
700:	test: 0.8374305	best: 0.8608986 (376)	total: 37.9s	remaining: 16.2s
800:	test: 0.8345703	best: 0.8608986 (376)	total: 43.3s	remaining: 10.8s
900:	test: 0.8367370	best: 0.8608986 (376)	total: 48.8s	remaining: 5.36s
999:	test: 0.8397132	best: 0.8608986 (376)	total: 54.2s	remaining: 0us

bestTest = 0.8608986316
bestIteration = 376

Shrink model to first 377 iterations.
target_7_1


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.6264676	best: 0.6264676 (0)	total: 99.9ms	remaining: 1m 39s
100:	test: 0.7918915	best: 0.7918915 (100)	total: 8.23s	remaining: 1m 13s
200:	test: 0.7967209	best: 0.7967209 (200)	total: 15.6s	remaining: 1m 1s
300:	test: 0.7985418	best: 0.7985418 (299)	total: 22.3s	remaining: 51.7s
400:	test: 0.7990088	best: 0.7990982 (385)	total: 29.3s	remaining: 43.7s
500:	test: 0.7999448	best: 0.7999508 (498)	total: 35.7s	remaining: 35.5s
600:	test: 0.8002022	best: 0.8003139 (579)	total: 42s	remaining: 27.9s
700:	test: 0.8005254	best: 0.8005639 (655)	total: 48.9s	remaining: 20.9s
800:	test: 0.8007262	best: 0.8008290 (719)	total: 56.6s	remaining: 14.1s
900:	test: 0.8009999	best: 0.8010396 (898)	total: 1m 3s	remaining: 6.97s
999:	test: 0.8012247	best: 0.8013741 (991)	total: 1m 10s	remaining: 0us

bestTest = 0.8013741247
bestIteration = 991

Shrink model to first 992 iterations.
target_7_2


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.5537767	best: 0.5537767 (0)	total: 115ms	remaining: 1m 55s
100:	test: 0.8292349	best: 0.8292349 (100)	total: 8.08s	remaining: 1m 11s
200:	test: 0.8350762	best: 0.8350768 (199)	total: 14.2s	remaining: 56.6s
300:	test: 0.8386743	best: 0.8387182 (296)	total: 20.4s	remaining: 47.3s
400:	test: 0.8399309	best: 0.8399649 (394)	total: 26.2s	remaining: 39.2s
500:	test: 0.8412984	best: 0.8412984 (500)	total: 33s	remaining: 32.9s
600:	test: 0.8424964	best: 0.8426220 (587)	total: 40.2s	remaining: 26.7s
700:	test: 0.8445703	best: 0.8445703 (700)	total: 47.2s	remaining: 20.1s
800:	test: 0.8447396	best: 0.8448742 (731)	total: 53.9s	remaining: 13.4s
900:	test: 0.8441994	best: 0.8448742 (731)	total: 1m	remaining: 6.64s
999:	test: 0.8441347	best: 0.8448742 (731)	total: 1m 6s	remaining: 0us

bestTest = 0.844874244
bestIteration = 731

Shrink model to first 732 iterations.
target_7_3


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.5294493	best: 0.5294493 (0)	total: 94.8ms	remaining: 1m 34s
100:	test: 0.7695887	best: 0.7696270 (95)	total: 6.26s	remaining: 55.7s
200:	test: 0.7716677	best: 0.7717697 (171)	total: 11.8s	remaining: 47.1s
300:	test: 0.7765705	best: 0.7765705 (300)	total: 17.6s	remaining: 40.9s
400:	test: 0.7761444	best: 0.7765705 (300)	total: 23.1s	remaining: 34.5s
500:	test: 0.7763901	best: 0.7774181 (484)	total: 28.7s	remaining: 28.6s
600:	test: 0.7756255	best: 0.7774181 (484)	total: 34.2s	remaining: 22.7s
700:	test: 0.7740697	best: 0.7774181 (484)	total: 39.9s	remaining: 17s
800:	test: 0.7724965	best: 0.7774181 (484)	total: 45.9s	remaining: 11.4s
900:	test: 0.7716171	best: 0.7774181 (484)	total: 52s	remaining: 5.71s
999:	test: 0.7695765	best: 0.7774181 (484)	total: 57.7s	remaining: 0us

bestTest = 0.7774180971
bestIteration = 484

Shrink model to first 485 iterations.
target_8_1


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.9368006	best: 0.9368006 (0)	total: 85.5ms	remaining: 1m 25s
100:	test: 0.9739294	best: 0.9739294 (100)	total: 6.81s	remaining: 1m
200:	test: 0.9749502	best: 0.9749689 (195)	total: 13.1s	remaining: 52.2s
300:	test: 0.9753695	best: 0.9753695 (300)	total: 19.2s	remaining: 44.6s
400:	test: 0.9755902	best: 0.9755917 (398)	total: 25.1s	remaining: 37.4s
500:	test: 0.9757716	best: 0.9757976 (478)	total: 31.2s	remaining: 31s
600:	test: 0.9760618	best: 0.9760618 (600)	total: 37.2s	remaining: 24.7s
700:	test: 0.9761487	best: 0.9761628 (693)	total: 43s	remaining: 18.3s
800:	test: 0.9762904	best: 0.9763039 (798)	total: 48.8s	remaining: 12.1s
900:	test: 0.9763275	best: 0.9763410 (887)	total: 54.9s	remaining: 6.03s
999:	test: 0.9764618	best: 0.9764646 (998)	total: 1m	remaining: 0us

bestTest = 0.9764645709
bestIteration = 998

Shrink model to first 999 iterations.
target_8_2


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.5003353	best: 0.5003353 (0)	total: 98ms	remaining: 1m 37s
100:	test: 0.8310688	best: 0.8310688 (100)	total: 7.24s	remaining: 1m 4s
200:	test: 0.8355544	best: 0.8355640 (199)	total: 14.1s	remaining: 56s
300:	test: 0.8372445	best: 0.8372683 (292)	total: 20.4s	remaining: 47.3s
400:	test: 0.8383668	best: 0.8383855 (389)	total: 27s	remaining: 40.3s
500:	test: 0.8390644	best: 0.8390956 (498)	total: 32.9s	remaining: 32.7s
600:	test: 0.8395576	best: 0.8396998 (592)	total: 38.8s	remaining: 25.8s
700:	test: 0.8395013	best: 0.8398610 (632)	total: 44.8s	remaining: 19.1s
800:	test: 0.8400574	best: 0.8400677 (798)	total: 51.4s	remaining: 12.8s
900:	test: 0.8405217	best: 0.8406160 (885)	total: 58s	remaining: 6.37s
999:	test: 0.8402388	best: 0.8407825 (926)	total: 1m 4s	remaining: 0us

bestTest = 0.8407824998
bestIteration = 926

Shrink model to first 927 iterations.
target_8_3


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.6865043	best: 0.6865043 (0)	total: 95.3ms	remaining: 1m 35s
100:	test: 0.8534456	best: 0.8534810 (98)	total: 6.72s	remaining: 59.8s
200:	test: 0.8588484	best: 0.8588484 (200)	total: 12.7s	remaining: 50.5s
300:	test: 0.8605337	best: 0.8605337 (300)	total: 18.6s	remaining: 43.3s
400:	test: 0.8625257	best: 0.8626607 (395)	total: 24.6s	remaining: 36.7s
500:	test: 0.8629742	best: 0.8630252 (497)	total: 30.7s	remaining: 30.6s
600:	test: 0.8630828	best: 0.8634252 (554)	total: 36.5s	remaining: 24.3s
700:	test: 0.8630180	best: 0.8634252 (554)	total: 42.5s	remaining: 18.1s
800:	test: 0.8632527	best: 0.8635103 (781)	total: 48.6s	remaining: 12.1s
900:	test: 0.8632640	best: 0.8635103 (781)	total: 54.6s	remaining: 5.99s
999:	test: 0.8632660	best: 0.8635493 (915)	total: 1m	remaining: 0us

bestTest = 0.8635492825
bestIteration = 915

Shrink model to first 916 iterations.
target_9_1


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.4880011	best: 0.4880011 (0)	total: 99.3ms	remaining: 1m 39s
100:	test: 0.7875712	best: 0.7875900 (99)	total: 5.98s	remaining: 53.2s
200:	test: 0.7946622	best: 0.7946622 (200)	total: 11.2s	remaining: 44.6s
300:	test: 0.7967067	best: 0.7968344 (280)	total: 16.6s	remaining: 38.4s
400:	test: 0.7965030	best: 0.7974688 (376)	total: 22s	remaining: 32.8s
500:	test: 0.7977806	best: 0.7992781 (463)	total: 27.3s	remaining: 27.2s
600:	test: 0.7979231	best: 0.7992781 (463)	total: 32.8s	remaining: 21.8s
700:	test: 0.7969306	best: 0.7992781 (463)	total: 38.1s	remaining: 16.3s
800:	test: 0.7961423	best: 0.7992781 (463)	total: 43.6s	remaining: 10.8s
900:	test: 0.7956505	best: 0.7992781 (463)	total: 49.1s	remaining: 5.4s
999:	test: 0.7959331	best: 0.7992781 (463)	total: 54.6s	remaining: 0us

bestTest = 0.7992780918
bestIteration = 463

Shrink model to first 464 iterations.
target_9_2


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.7422151	best: 0.7422151 (0)	total: 94.2ms	remaining: 1m 34s
100:	test: 0.8180757	best: 0.8181328 (96)	total: 6.74s	remaining: 60s
200:	test: 0.8207219	best: 0.8207620 (197)	total: 12.7s	remaining: 50.7s
300:	test: 0.8212954	best: 0.8214204 (276)	total: 18.8s	remaining: 43.6s
400:	test: 0.8217293	best: 0.8220027 (376)	total: 25.2s	remaining: 37.7s
500:	test: 0.8217780	best: 0.8221471 (484)	total: 31.3s	remaining: 31.1s
600:	test: 0.8217844	best: 0.8221471 (484)	total: 37.3s	remaining: 24.7s
700:	test: 0.8214192	best: 0.8221471 (484)	total: 43.2s	remaining: 18.4s
800:	test: 0.8214592	best: 0.8221471 (484)	total: 49.2s	remaining: 12.2s
900:	test: 0.8210459	best: 0.8221471 (484)	total: 55s	remaining: 6.05s
999:	test: 0.8205917	best: 0.8221471 (484)	total: 1m	remaining: 0us

bestTest = 0.8221470793
bestIteration = 484

Shrink model to first 485 iterations.
target_9_3


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.5106392	best: 0.5106392 (0)	total: 99ms	remaining: 1m 38s
100:	test: 0.6782177	best: 0.6784668 (99)	total: 6.38s	remaining: 56.8s
200:	test: 0.6819357	best: 0.6819357 (200)	total: 12s	remaining: 47.8s
300:	test: 0.6824903	best: 0.6824945 (297)	total: 17.5s	remaining: 40.5s
400:	test: 0.6827384	best: 0.6832131 (344)	total: 23.1s	remaining: 34.5s
500:	test: 0.6836767	best: 0.6840187 (464)	total: 28.7s	remaining: 28.6s
600:	test: 0.6840498	best: 0.6844542 (574)	total: 34.3s	remaining: 22.8s
700:	test: 0.6845856	best: 0.6848381 (659)	total: 39.9s	remaining: 17s
800:	test: 0.6843530	best: 0.6849474 (729)	total: 45.7s	remaining: 11.3s
900:	test: 0.6818343	best: 0.6849474 (729)	total: 51.4s	remaining: 5.64s
999:	test: 0.6801213	best: 0.6849474 (729)	total: 57.1s	remaining: 0us

bestTest = 0.6849474046
bestIteration = 729

Shrink model to first 730 iterations.
target_9_4


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.5729996	best: 0.5729996 (0)	total: 91.7ms	remaining: 1m 31s
100:	test: 0.9039523	best: 0.9039523 (100)	total: 6.16s	remaining: 54.8s
200:	test: 0.9059828	best: 0.9069179 (177)	total: 11.5s	remaining: 45.7s
300:	test: 0.9075935	best: 0.9081602 (287)	total: 17s	remaining: 39.4s
400:	test: 0.9056407	best: 0.9081602 (287)	total: 22.4s	remaining: 33.5s
500:	test: 0.9053367	best: 0.9081602 (287)	total: 27.9s	remaining: 27.8s
600:	test: 0.9049835	best: 0.9081602 (287)	total: 34s	remaining: 22.6s
700:	test: 0.9050454	best: 0.9081602 (287)	total: 40s	remaining: 17.1s
800:	test: 0.9036354	best: 0.9081602 (287)	total: 46s	remaining: 11.4s
900:	test: 0.9018943	best: 0.9081602 (287)	total: 51.6s	remaining: 5.67s
999:	test: 0.8997527	best: 0.9081602 (287)	total: 58.2s	remaining: 0us

bestTest = 0.9081602262
bestIteration = 287

Shrink model to first 288 iterations.
target_9_5


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.5452491	best: 0.5452491 (0)	total: 85.4ms	remaining: 1m 25s
100:	test: 0.8228690	best: 0.8230737 (96)	total: 7.15s	remaining: 1m 3s
200:	test: 0.8248413	best: 0.8248490 (199)	total: 13.4s	remaining: 53.1s
300:	test: 0.8242486	best: 0.8252209 (212)	total: 18.9s	remaining: 43.8s
400:	test: 0.8232934	best: 0.8252209 (212)	total: 24.4s	remaining: 36.4s
500:	test: 0.8228459	best: 0.8252209 (212)	total: 30s	remaining: 29.8s
600:	test: 0.8233988	best: 0.8252209 (212)	total: 35.6s	remaining: 23.6s
700:	test: 0.8219862	best: 0.8252209 (212)	total: 41.3s	remaining: 17.6s
800:	test: 0.8207707	best: 0.8252209 (212)	total: 47s	remaining: 11.7s
900:	test: 0.8184751	best: 0.8252209 (212)	total: 53s	remaining: 5.82s
999:	test: 0.8169591	best: 0.8252209 (212)	total: 58.8s	remaining: 0us

bestTest = 0.8252209236
bestIteration = 212

Shrink model to first 213 iterations.
target_9_6


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.6280266	best: 0.6280266 (0)	total: 110ms	remaining: 1m 49s
100:	test: 0.6786619	best: 0.6786619 (100)	total: 8.26s	remaining: 1m 13s
200:	test: 0.6824884	best: 0.6825608 (197)	total: 15s	remaining: 59.8s
300:	test: 0.6834275	best: 0.6834417 (299)	total: 21.3s	remaining: 49.6s
400:	test: 0.6836961	best: 0.6838152 (366)	total: 27.6s	remaining: 41.3s
500:	test: 0.6840788	best: 0.6841126 (493)	total: 33.9s	remaining: 33.8s
600:	test: 0.6838377	best: 0.6841629 (519)	total: 40s	remaining: 26.6s
700:	test: 0.6840760	best: 0.6841820 (691)	total: 46.2s	remaining: 19.7s
800:	test: 0.6840019	best: 0.6841999 (734)	total: 52.4s	remaining: 13s
900:	test: 0.6840988	best: 0.6841999 (734)	total: 58.8s	remaining: 6.46s
999:	test: 0.6841106	best: 0.6841999 (734)	total: 1m 5s	remaining: 0us

bestTest = 0.6841998698
bestIteration = 734

Shrink model to first 735 iterations.
target_9_7


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.6611838	best: 0.6611838 (0)	total: 108ms	remaining: 1m 48s
100:	test: 0.7461177	best: 0.7461177 (100)	total: 7.68s	remaining: 1m 8s
200:	test: 0.7485284	best: 0.7486160 (195)	total: 14.5s	remaining: 57.8s
300:	test: 0.7494334	best: 0.7494780 (293)	total: 21.2s	remaining: 49.3s
400:	test: 0.7502272	best: 0.7502424 (398)	total: 28.1s	remaining: 41.9s
500:	test: 0.7500271	best: 0.7502424 (398)	total: 34.6s	remaining: 34.4s
600:	test: 0.7498211	best: 0.7502548 (525)	total: 41.8s	remaining: 27.8s
700:	test: 0.7494979	best: 0.7502548 (525)	total: 48.6s	remaining: 20.7s
800:	test: 0.7489932	best: 0.7502548 (525)	total: 54.8s	remaining: 13.6s
900:	test: 0.7487515	best: 0.7502548 (525)	total: 1m 1s	remaining: 6.72s
999:	test: 0.7489720	best: 0.7502548 (525)	total: 1m 7s	remaining: 0us

bestTest = 0.7502547977
bestIteration = 525

Shrink model to first 526 iterations.
target_9_8


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.5704905	best: 0.5704905 (0)	total: 88.7ms	remaining: 1m 28s
100:	test: 0.9172480	best: 0.9172480 (100)	total: 7.04s	remaining: 1m 2s
200:	test: 0.9198825	best: 0.9198866 (197)	total: 12.9s	remaining: 51.2s
300:	test: 0.9209638	best: 0.9210579 (291)	total: 18.7s	remaining: 43.4s
400:	test: 0.9220060	best: 0.9220119 (394)	total: 24.6s	remaining: 36.7s
500:	test: 0.9225488	best: 0.9225612 (499)	total: 31.7s	remaining: 31.5s
600:	test: 0.9228431	best: 0.9228996 (551)	total: 38.7s	remaining: 25.7s
700:	test: 0.9232541	best: 0.9233435 (696)	total: 45.2s	remaining: 19.3s
800:	test: 0.9240014	best: 0.9240689 (787)	total: 51.6s	remaining: 12.8s
900:	test: 0.9241177	best: 0.9242638 (867)	total: 57.7s	remaining: 6.34s
999:	test: 0.9246701	best: 0.9246917 (989)	total: 1m 3s	remaining: 0us

bestTest = 0.9246917125
bestIteration = 989

Shrink model to first 990 iterations.
target_10_1


MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

Learning rate set to 0.159826
0:	test: 0.7031518	best: 0.7031518 (0)	total: 113ms	remaining: 1m 53s
100:	test: 0.7394039	best: 0.7394039 (100)	total: 7.73s	remaining: 1m 8s
200:	test: 0.7434267	best: 0.7434428 (198)	total: 15.2s	remaining: 1m
300:	test: 0.7449829	best: 0.7449829 (300)	total: 22.2s	remaining: 51.6s
400:	test: 0.7455434	best: 0.7455636 (387)	total: 29.2s	remaining: 43.7s
500:	test: 0.7462957	best: 0.7463095 (499)	total: 36.2s	remaining: 36.1s
600:	test: 0.7463355	best: 0.7463513 (566)	total: 43.1s	remaining: 28.6s
700:	test: 0.7465923	best: 0.7465924 (695)	total: 50.4s	remaining: 21.5s
800:	test: 0.7468680	best: 0.7469202 (776)	total: 57.2s	remaining: 14.2s
900:	test: 0.7470894	best: 0.7471569 (864)	total: 1m 3s	remaining: 6.99s
999:	test: 0.7470772	best: 0.7471569 (864)	total: 1m 9s	remaining: 0us

bestTest = 0.747156876
bestIteration = 864

Shrink model to first 865 iterations.
CPU times: user 6h 37min 54s, sys: 43min 34s, total: 7h 21min 29s
Wall time: 41min 22s


**Windows:**
- CPU times: user 14min 14s, sys: 6min 44s, total: 20min 59s
- Wall time: 9min 41s

**Mac:**
- CPU times: user 6h 37min 54s, sys: 43min 34s, total: 7h 21min 29s
- Wall time: 41min 22s

In [39]:
test = pl.read_parquet(data_path+'test/'+'test_main_features.parquet'
                      ).fill_null(-99)

test = test.with_columns(
    pl.col(cat_feature_names).cast(pl.Int32)
)
test

customer_id,cat_feature_1,cat_feature_2,cat_feature_3,cat_feature_4,cat_feature_5,cat_feature_6,cat_feature_7,cat_feature_8,cat_feature_9,cat_feature_10,cat_feature_11,cat_feature_12,cat_feature_13,cat_feature_14,cat_feature_15,cat_feature_16,cat_feature_17,cat_feature_18,cat_feature_19,cat_feature_20,cat_feature_21,cat_feature_22,cat_feature_23,cat_feature_24,cat_feature_25,cat_feature_26,cat_feature_27,cat_feature_28,cat_feature_29,cat_feature_30,cat_feature_31,cat_feature_32,cat_feature_33,cat_feature_34,cat_feature_35,cat_feature_36,…,num_feature_96,num_feature_97,num_feature_98,num_feature_99,num_feature_100,num_feature_101,num_feature_102,num_feature_103,num_feature_104,num_feature_105,num_feature_106,num_feature_107,num_feature_108,num_feature_109,num_feature_110,num_feature_111,num_feature_112,num_feature_113,num_feature_114,num_feature_115,num_feature_116,num_feature_117,num_feature_118,num_feature_119,num_feature_120,num_feature_121,num_feature_122,num_feature_123,num_feature_124,num_feature_125,num_feature_126,num_feature_127,num_feature_128,num_feature_129,num_feature_130,num_feature_131,num_feature_132
i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,…,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64
1750001,1,0,2,0,2,3,2,2,4,0,0,0,0,1,0,0,2,2,2,0,0,0,2,0,0,0,1,2,1,2,2,0,0,212,0,0,…,0.503368,-0.264397,-0.004499,-0.150243,-0.241966,-0.002297,-0.043592,-0.061488,-0.383148,-0.009552,0.0,-0.26649,-0.060492,-0.279105,-99.0,-0.429813,-99.0,-0.009654,-0.293036,-99.0,-0.493959,-0.01556,-99.0,-99.0,-99.0,-99.0,-0.001357,-0.031281,-0.046146,0.062409,-0.475778,-99.0,-99.0,0.018451,-0.765018,-0.397803,-0.373734
1750002,0,0,2,0,2,3,2,2,4,0,0,0,0,1,0,0,2,2,2,1,0,0,2,1,0,0,1,2,1,2,2,1,0,212,1,0,…,-0.284519,-0.264397,-0.004278,0.386748,-0.620003,-0.002297,-0.043592,0.367167,-0.025938,-99.0,0.0,-0.26649,-0.060492,-0.279105,-99.0,-0.429813,-99.0,-0.009654,3.569151,-99.0,-0.57313,-0.015727,-99.0,-99.0,-0.3427,-99.0,-0.001357,-0.031281,-0.046146,-99.0,-0.475778,-99.0,-99.0,-0.170724,-0.357487,-0.397803,-0.373734
1750003,1,0,2,1,2,3,2,2,4,2,0,0,0,1,2,0,2,2,2,0,0,0,2,0,0,0,1,2,1,2,2,1,0,212,0,0,…,-99.0,-0.264397,-0.004499,-99.0,-99.0,-0.002297,-99.0,-0.061488,-0.264078,-0.009468,-99.0,-0.26649,-0.060492,-0.279105,-99.0,-99.0,-99.0,-0.009654,-99.0,-99.0,-99.0,-0.01934,-99.0,0.022101,-99.0,-99.0,-99.0,-0.031281,-99.0,-99.0,-0.455446,-99.0,-99.0,0.018451,-0.785394,-99.0,-99.0
1750004,1,0,2,0,2,3,2,2,4,2,0,0,0,1,2,0,2,2,2,0,0,2,2,0,0,0,1,2,1,2,2,0,0,212,0,0,…,-0.284519,-99.0,-0.004499,-0.150243,-99.0,-0.002297,-0.043592,-0.061488,-0.383148,-99.0,0.0,-99.0,-0.060492,-99.0,-99.0,-0.429813,-99.0,-0.009654,-0.293036,-99.0,-0.018932,-0.015257,-99.0,-99.0,-0.388464,-99.0,-0.001357,-0.031281,-0.046146,0.054572,-0.475778,-99.0,-99.0,-99.0,-0.520499,-99.0,-99.0
1750005,0,1,2,0,2,3,2,2,4,2,0,0,0,1,2,0,2,2,2,1,0,0,2,1,1,0,1,2,1,2,2,1,0,212,0,0,…,-0.284519,-0.264397,-0.004499,-0.150243,1.663358,-0.002297,-0.043592,-0.061488,-0.383148,-99.0,0.0,-0.26649,-0.060492,-0.279105,-99.0,-0.429813,-99.0,-0.009654,-0.293036,-99.0,-0.57313,-0.017804,-99.0,-0.24167,-99.0,-0.138239,-0.001357,-0.031281,-0.046146,-99.0,1.533778,1.318607,-99.0,-0.170724,0.314939,-99.0,-99.0
…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…
1999996,1,2,2,1,2,3,2,2,4,2,2,2,2,1,2,0,2,2,2,2,2,2,2,2,2,2,2,2,1,2,2,2,2,212,2,2,…,-99.0,-99.0,-0.004499,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0,-99.0
1999997,1,2,0,1,0,3,0,0,4,1,2,2,2,0,0,0,0,0,0,2,2,0,0,2,2,2,2,0,0,0,0,2,2,212,2,2,…,-0.284519,-0.264397,-0.004499,-99.0,-0.761335,-0.002297,-0.043592,-

## Предсказание модели

In [40]:
test_pool = Pool(data = test.drop("customer_id").to_pandas(), 
                 #cat_features = cat_feature_names
                )
test_pool

Заметим, что колонки `target` почему-то разделены на группы:
|       | target_1 | taget_2 | target_3 | target_4 | target_5 | target_6 | taget_7 | taget_8 | taget_9 | target_10 |
|-------|----------|---------|----------|----------|----------|----------|---------|---------|---------|-----------|
| count | 5        | 8       | 5        | 1        | 2        | 5        | 3       | 3       | 8       | 1         |


In [41]:
for tcols in target_columns:
    print(tcols)
    model.load_model(f"wocat{tcols}.cbm")
    test_predict = model.predict(test_pool, prediction_type = "RawFormulaVal")
    test = test.with_columns(temp=test_predict).rename({"temp":tcols})
test

target_1_1
target_1_2
target_1_3
target_1_4
target_1_5
target_2_1
target_2_2
target_2_3
target_2_4
target_2_5
target_2_6
target_2_7
target_2_8
target_3_1
target_3_2
target_3_3
target_3_4
target_3_5
target_4_1
target_5_1
target_5_2
target_6_1
target_6_2
target_6_3
target_6_4
target_6_5
target_7_1
target_7_2
target_7_3
target_8_1
target_8_2
target_8_3
target_9_1
target_9_2
target_9_3
target_9_4
target_9_5
target_9_6
target_9_7
target_9_8
target_10_1


customer_id,cat_feature_1,cat_feature_2,cat_feature_3,cat_feature_4,cat_feature_5,cat_feature_6,cat_feature_7,cat_feature_8,cat_feature_9,cat_feature_10,cat_feature_11,cat_feature_12,cat_feature_13,cat_feature_14,cat_feature_15,cat_feature_16,cat_feature_17,cat_feature_18,cat_feature_19,cat_feature_20,cat_feature_21,cat_feature_22,cat_feature_23,cat_feature_24,cat_feature_25,cat_feature_26,cat_feature_27,cat_feature_28,cat_feature_29,cat_feature_30,cat_feature_31,cat_feature_32,cat_feature_33,cat_feature_34,cat_feature_35,cat_feature_36,…,target_1_5,target_2_1,target_2_2,target_2_3,target_2_4,target_2_5,target_2_6,target_2_7,target_2_8,target_3_1,target_3_2,target_3_3,target_3_4,target_3_5,target_4_1,target_5_1,target_5_2,target_6_1,target_6_2,target_6_3,target_6_4,target_6_5,target_7_1,target_7_2,target_7_3,target_8_1,target_8_2,target_8_3,target_9_1,target_9_2,target_9_3,target_9_4,target_9_5,target_9_6,target_9_7,target_9_8,target_10_1
i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,i32,…,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64
1750001,1,0,2,0,2,3,2,2,4,0,0,0,0,1,0,0,2,2,2,0,0,0,2,0,0,0,1,2,1,2,2,0,0,212,0,0,…,-7.029726,-4.984127,-5.272237,-7.648119,-4.768143,-6.958062,-6.138118,-8.929741,-12.500473,-2.240438,-4.582098,-7.585328,-8.788304,-11.427572,-4.502799,-5.568636,-6.659411,-6.101653,-6.153649,-5.613236,-5.934041,-9.173555,-2.753256,-4.745554,-5.110236,-6.248015,-4.634518,-3.979078,-5.550732,-2.905332,-4.018138,-10.36281,-7.443879,-0.295655,-2.368205,-7.109498,-0.59118
1750002,0,0,2,0,2,3,2,2,4,0,0,0,0,1,0,0,2,2,2,1,0,0,2,1,0,0,1,2,1,2,2,1,0,212,1,0,…,-6.610275,-3.243938,-5.277364,-7.327784,-4.304502,-5.916581,-6.39367,-8.072554,-12.004719,-2.78021,-5.05643,-7.289385,-8.92129,-11.614137,-6.261377,-5.052025,-6.432152,-6.026409,-5.267215,-5.941416,-5.997136,-9.000745,-3.318007,-4.060681,-6.220353,-4.872623,-2.352105,-5.616156,-6.227298,-1.540115,-3.646324,-4.635172,-4.257187,-0.754386,-1.90217,-9.887006,-1.284431
1750003,1,0,2,1,2,3,2,2,4,2,0,0,0,1,2,0,2,2,2,0,0,0,2,0,0,0,1,2,1,2,2,1,0,212,0,0,…,-6.814286,-4.909994,-5.376739,-7.487052,-5.326527,-6.483825,-5.432946,-9.442807,-13.571656,-2.074231,-4.420578,-7.045689,-7.602565,-11.253423,-4.028619,-4.881729,-7.984861,-5.044735,-6.333924,-6.462128,-5.457805,-9.109361,-8.434558,-4.340428,-6.308007,-3.500106,-3.557577,-4.335899,-6.04867,-3.293171,-4.357157,-10.874541,-8.247507,-1.350193,-3.740605,-7.833687,-0.436388
1750004,1,0,2,0,2,3,2,2,4,2,0,0,0,1,2,0,2,2,2,0,0,2,2,0,0,0,1,2,1,2,2,0,0,212,0,0,…,-9.349048,-5.171648,-5.341965,-7.851793,-4.644168,-8.172426,-6.484761,-9.873532,-13.317555,-3.120563,-4.404931,-7.442555,-8.799005,-11.117671,-6.310752,-5.769997,-6.847197,-6.182241,-5.586413,-5.510683,-6.574431,-10.58214,-2.812452,-4.797092,-5.467015,-6.521562,-5.397478,-3.660425,-6.213749,-2.886229,-3.730834,-7.399069,-5.814878,-0.486924,-2.036077,-6.79542,-0.633722
1750005,0,1,2,0,2,3,2,2,4,2,0,0,0,1,2,0,2,2,2,1,0,0,2,1,1,0,1,2,1,2,2,1,0,212,0,0,…,-7.971654,-4.688121,-5.452617,-7.410337,-5.793812,-6.067568,-5.885282,-10.019301,-12.301235,-2.264424,-4.759583,-7.120027,-8.689535,-12.003541,-6.707885,-5.882269,-6.376899,-4.427032,-4.90637,-6.94809,-6.309877,-9.398426,-8.146391,-4.637849,-6.604285,-5.710643,-3.241156,-5.179743,-5.612338,-2.624685,-3.618327,-8.062036,-4.580536,-1.560034,-2.087354,-8.707927,-0.612452
…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…
1999996,1,2,2,1,2,3,2,2,4,2,2,2,2,1,2,0,2,2,2,2,2,2,2,2,2,2,2,2,1,2,2,2,2,212,2,2,…,-8.377092,-7.672467,-6.800115,-8.747928,-5.204561,-7.206981,-5.240312,-11.650663,-12.930963,-3.06933,-5.097379,-7.678875,-8.817793,-10.696985,-7.72888,-5.965295,-7.120524,-4.769619,-4.740723,-5.59654,-6.11496,-9.663275,-3.224974,-4.53362,-6.46

## Формирование submission

In [42]:
predict_schema = ["customer_id"]+[col.replace("target_", "predict_") for col in target.columns if col.startswith("target_")]

catboost_predictions = pl.DataFrame(test.select(target.columns), schema = predict_schema)
catboost_predictions

customer_id,predict_1_1,predict_1_2,predict_1_3,predict_1_4,predict_1_5,predict_2_1,predict_2_2,predict_2_3,predict_2_4,predict_2_5,predict_2_6,predict_2_7,predict_2_8,predict_3_1,predict_3_2,predict_3_3,predict_3_4,predict_3_5,predict_4_1,predict_5_1,predict_5_2,predict_6_1,predict_6_2,predict_6_3,predict_6_4,predict_6_5,predict_7_1,predict_7_2,predict_7_3,predict_8_1,predict_8_2,predict_8_3,predict_9_1,predict_9_2,predict_9_3,predict_9_4,predict_9_5,predict_9_6,predict_9_7,predict_9_8,predict_10_1
i32,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64,f64
1750001,-7.188548,-6.073806,-4.952836,-4.390186,-7.029726,-4.984127,-5.272237,-7.648119,-4.768143,-6.958062,-6.138118,-8.929741,-12.500473,-2.240438,-4.582098,-7.585328,-8.788304,-11.427572,-4.502799,-5.568636,-6.659411,-6.101653,-6.153649,-5.613236,-5.934041,-9.173555,-2.753256,-4.745554,-5.110236,-6.248015,-4.634518,-3.979078,-5.550732,-2.905332,-4.018138,-10.36281,-7.443879,-0.295655,-2.368205,-7.109498,-0.59118
1750002,-5.429091,-5.432858,-3.615,-4.007675,-6.610275,-3.243938,-5.277364,-7.327784,-4.304502,-5.916581,-6.39367,-8.072554,-12.004719,-2.78021,-5.05643,-7.289385,-8.92129,-11.614137,-6.261377,-5.052025,-6.432152,-6.026409,-5.267215,-5.941416,-5.997136,-9.000745,-3.318007,-4.060681,-6.220353,-4.872623,-2.352105,-5.616156,-6.227298,-1.540115,-3.646324,-4.635172,-4.257187,-0.754386,-1.90217,-9.887006,-1.284431
1750003,-5.547932,-5.512655,-3.989439,-3.141071,-6.814286,-4.909994,-5.376739,-7.487052,-5.326527,-6.483825,-5.432946,-9.442807,-13.571656,-2.074231,-4.420578,-7.045689,-7.602565,-11.253423,-4.028619,-4.881729,-7.984861,-5.044735,-6.333924,-6.462128,-5.457805,-9.109361,-8.434558,-4.340428,-6.308007,-3.500106,-3.557577,-4.335899,-6.04867,-3.293171,-4.357157,-10.874541,-8.247507,-1.350193,-3.740605,-7.833687,-0.436388
1750004,-7.566135,-7.603064,-5.278802,-5.239091,-9.349048,-5.171648,-5.341965,-7.851793,-4.644168,-8.172426,-6.484761,-9.873532,-13.317555,-3.120563,-4.404931,-7.442555,-8.799005,-11.117671,-6.310752,-5.769997,-6.847197,-6.182241,-5.586413,-5.510683,-6.574431,-10.58214,-2.812452,-4.797092,-5.467015,-6.521562,-5.397478,-3.660425,-6.213749,-2.886229,-3.730834,-7.399069,-5.814878,-0.486924,-2.036077,-6.79542,-0.633722
1750005,-4.27988,-5.977506,-3.461046,-3.400909,-7.971654,-4.688121,-5.452617,-7.410337,-5.793812,-6.067568,-5.885282,-10.019301,-12.301235,-2.264424,-4.759583,-7.120027,-8.689535,-12.003541,-6.707885,-5.882269,-6.376899,-4.427032,-4.90637,-6.94809,-6.309877,-9.398426,-8.146391,-4.637849,-6.604285,-5.710643,-3.241156,-5.179743,-5.612338,-2.624685,-3.618327,-8.062036,-4.580536,-1.560034,-2.087354,-8.707927,-0.612452
…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…
1999996,-6.051311,-6.874092,-5.995539,-5.042032,-8.377092,-7.672467,-6.800115,-8.747928,-5.204561,-7.206981,-5.240312,-11.650663,-12.930963,-3.06933,-5.097379,-7.678875,-8.817793,-10.696985,-7.72888,-5.965295,-7.120524,-4.769619,-4.740723,-5.59654,-6.11496,-9.663275,-3.224974,-4.53362,-6.4663,-5.602787,-3.564221,-5.912134,-8.503816,-6.413233,-4.249313,-5.525507,-6.638898,-2.108278,-5.594189,-8.196369,1.060799
1999997,-6.529483,-6.389601,-5.011321,-5.018652,-6.680862,-4.547423,-6.606545,-4.735188,-6.332781,-6.847538,-7.423078,-7.826604,-7.426893,-1.534622,-0.759503,-7.413406,-5.877029,-6.830706,-5.102887,-3.827651,-7.40628,-4.365667,-5.011363,-5.322079,-3.856538,-6.294744,-2.710852,-5.312536,-6.660706,-0.822982,-3.792812,-5.766291,-7.093556,-4.048414,-5.45185,-8.275241,-7.298131,-2.78043,-3.284586,-4.113454,-1.826927
1999998,-7.934823,-8.545553,-8.029762,-7.286466,-11.611219,-7.173125,-10.392511,-7.055096,-4.916588,-7.036708,-5.854536,-10.584952,-13.311935,-2.592671,-0.352829,-8.054361,-8.706239,-4.310983,-7.191545,-4.865198,-6.586192,-3.968004,-6.597627,-5.606631,-3.419872,-8.668968,-4.426738,-7.335197,-7.401307,1.902676,-5.559064,-5.067

## Лучшие и худшие предсказанные категории

In [44]:
import glob
import os

results = []
model_files = glob.glob("wocat*.cbm")

for model_file in model_files:
    target_name = model_file.replace("wocat", "").replace(".cbm", "")
    
    model = CatBoostClassifier()
    model.load_model(model_file)
    
    best_iteration = model.get_best_iteration()
    
    best_score = model.get_best_score()
    
    auc_value = None
    if best_score and 'validation' in best_score:
        if 'AUC' in best_score['validation']:
            auc_value = best_score['validation']['AUC']
        else:
            auc_value = list(best_score['validation'].values())[0]
    
    results.append({
        'target_column': target_name,
        'best_iteration': best_iteration,
        'best_AUC': auc_value,
        'model_file': model_file
    })

results_df = pd.DataFrame(results)
results_df = results_df.sort_values('best_AUC', ascending=False).reset_index(drop=True)
results_df['rank'] = results_df.index + 1

print("\n" + "="*80)
print("ИТОГОВАЯ ТАБЛИЦА РЕЗУЛЬТАТОВ (из .cbm файлов)")
print("="*80)
print(results_df.to_string(index=False))

results_df.to_csv('ind_clas_41_0818_results.csv', index=False)
print("\nРезультаты сохранены в 'ind_clas_41_0818_results.csv'")

print("\n" + "="*80)
print("РЕЗУЛЬТАТЫ (форматированные)")
print("="*80)
for _, row in results_df.iterrows():
    print(f"{row['target_column']:<20} | "
          f"Итерация: {row['best_iteration']:<6} | "
          f"AUC: {row['best_AUC']:.6f} | "
          f"Ранг: {row['rank']}")


ИТОГОВАЯ ТАБЛИЦА РЕЗУЛЬТАТОВ (из .cbm файлов)
target_column  best_iteration  best_AUC           model_file  rank
   target_3_5             948  0.976706  wocattarget_3_5.cbm     1
   target_8_1             998  0.976465  wocattarget_8_1.cbm     2
   target_2_8              25  0.975721  wocattarget_2_8.cbm     3
   target_2_7             362  0.939787  wocattarget_2_7.cbm     4
   target_3_4             778  0.938447  wocattarget_3_4.cbm     5
   target_9_8             989  0.924692  wocattarget_9_8.cbm     6
   target_2_2             941  0.922317  wocattarget_2_2.cbm     7
   target_1_1             808  0.917369  wocattarget_1_1.cbm     8
   target_9_4             287  0.908160  wocattarget_9_4.cbm     9
   target_3_2             911  0.906357  wocattarget_3_2.cbm    10
   target_1_3             999  0.867878  wocattarget_1_3.cbm    11
   target_1_5             799  0.867506  wocattarget_1_5.cbm    12
   target_8_3             915  0.863549  wocattarget_8_3.cbm    13
   target_6_5  

target_column        | best_iteration   | best_AUC      | rank
---------------------|------------------|---------------|----------
target_3_5           | 948    | 0.976706 | 1
target_8_1           | 998    | 0.976465 | 2
target_2_8           | 25     | 0.975721 | 3
target_2_7           | 362    | 0.939787 | 4
target_3_4           | 778    | 0.938447 | 5
target_9_8           | 989    | 0.924692 | 6
target_2_2           | 941    | 0.922317 | 7
target_1_1           | 808    | 0.917369 | 8
target_9_4           | 287    | 0.908160 | 9
target_3_2           | 911    | 0.906357 | 10
target_1_3           | 999    | 0.867878 | 11
target_1_5           | 799    | 0.867506 | 12
target_8_3           | 915    | 0.863549 | 13
target_6_5           | 376    | 0.860899 | 14
target_4_1           | 504    | 0.857183 | 15
target_6_4           | 157    | 0.851358 | 16
target_7_2           | 731    | 0.844874 | 17
target_8_2           | 926    | 0.840782 | 18
target_1_4           | 723    | 0.825604 | 19
target_9_5           | 212    | 0.825221 | 20
target_9_2           | 484    | 0.822147 | 21
target_1_2           | 205    | 0.807214 | 22
target_2_3           | 101    | 0.805713 | 23
target_7_1           | 991    | 0.801374 | 24
target_9_1           | 463    | 0.799278 | 25
target_2_1           | 378    | 0.795340 | 26
target_3_3           | 205    | 0.783547 | 27
target_7_3           | 484    | 0.777418 | 28
target_2_5           | 696    | 0.776152 | 29
target_6_3           | 378    | 0.760624 | 30
target_9_7           | 525    | 0.750255 | 31
target_10_1          | 864    | 0.747157 | 32
target_2_6           | 607    | 0.740591 | 33
target_5_1           | 283    | 0.737932 | 34
target_2_4           | 139    | 0.728115 | 35
target_6_2           | 945    | 0.723372 | 36
target_5_2           | 593    | 0.712871 | 37
target_6_1           | 692    | 0.696081 | 38
target_3_1           | 884    | 0.687275 | 39
target_9_3           | 729    | 0.684947 | 40
target_9_6           | 734    | 0.684200 | 41

Вывод: надо получше поработать над таргетами `target_9_6`, `target_9_3`, `target_3_1`, ... У них ROC-AUC < 0.7, что сильно занижает результативность нашего решения.

Либо попытаться улучшить модель `CatBoost`, либо выбрать для них другую.

Надо посмотреть на эти таргеты под прицелом: вдруг они встречаются в `train` единажды или типо того. То есть мы банально не можем на них обучиться, записей слишком мало (?)